actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -474,6 +474,247 @@ def apply_form_for_options!(record, object, options) #:nodoc:
+      #     ...
+      #   end
+      #
+      # generates
+      #
+      #   <form action="http://www.example.com" method="post" data-behavior="autosave" name="go">
+      #     <input name="_method" type="hidden" value="patch" />
+      #     ...
+      #   </form>
+      #
+      # === Removing hidden model id's
+      #
+      # The +form_with+ method automatically includes the model id as a hidden field in the form.
+      # This is used to maintain the correlation between the form data and its associated model.
+      # Some ORM systems do not use IDs on nested models so in this case you want to be able
+      # to disable the hidden id.
+      #
+      # In the following example the Post model has many Comments stored within it in a NoSQL database,
+      # thus there is no primary key for comments.
+      #
+      #   form_with(model: @post) do |form|
+      #     form.fields(:comments, include_id: false) do |fields|
+      #       ...
+      #     end
+      #   end
+      #
+      # === Customized form builders
+      #
+      # You can also build forms using a customized FormBuilder class. Subclass
+      # FormBuilder and override or define some more helpers, then use your
+      # custom builder. For example, let's say you made a helper to
+      # automatically add labels to form inputs.
+      #
+      #   form_with model: @person, url: { action: "create" }, builder: LabellingFormBuilder do |form|
+      #     form.text_field :first_name
+      #     form.text_field :last_name
+      #     form.text_area :biography
+      #     form.check_box :admin
+      #     form.submit
+      #   end
+      #
+      # In this case, if you use:
+      #
+      #   <%= render form %>
+      #
+      # The rendered template is <tt>people/_labelling_form</tt> and the local
+      # variable referencing the form builder is called
+      # <tt>labelling_form</tt>.
+      #
+      # The custom FormBuilder class is automatically merged with the options
+      # of a nested +fields+ call, unless it's explicitly set.
+      #
+      # In many cases you will want to wrap the above in another helper, so you
+      # could do something like the following:
+      #
+      #   def labelled_form_for(**options, &block)
+      #     form_with(**options.merge(builder: LabellingFormBuilder), &block)
+      #   end
+      def form_with(model: nil, scope: nil, url: nil, format: nil, html: {}, remote: true, **options)
+        if model
+          url ||= polymorphic_path(model, format: format)
+
+          model   = model.last if model.is_a?(Array)
+          scope ||= model_name_from_record_or_class(model).param_key
+        end
+
+        html_options = html.merge(options.except(:index, :include_id, :builder))
+        html_options[:method] ||= :patch if model.respond_to?(:persisted?) && model.persisted?
+        html_options[:remote] = remote unless html_options.key?(:remote)
```

Comment:
It's used in the `else` too as the last parameter here too: `html_options = html_options_for_form(url || {}, html_options)`

But yeah, let me see what I can do with the composition 👍